src/Compression/CompressionCodecDeflateQpl.cpp

```diff
@@ -362,8 +332,14 @@ void CompressionCodecDeflateQpl::doDecompressData(const char * source, uint32_t
         {
             int32_t res = HardwareCodecDeflateQpl::RET_ERROR;
             if (DeflateQplJobHWPool::instance().isJobPoolReady())
-                res = hw_codec->doDecompressDataSynchronous(source, source_size, dest, uncompressed_size);
-            if (res == HardwareCodecDeflateQpl::RET_ERROR)
+            {
+                res = hw_codec->doDecompressDataAsynchronous(source, source_size, dest, uncompressed_size);
```

Comment:
Hmmm, if there were already earlier, unrelated jobs in the hardware queue, then the flush in l. 338 will wait for them even if the original thread which scheduled these job hasn't issued the flush yet?

Reply:
Here let me restore the function doDecompressDataSynchronous, I think it would be more reasonable.